File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    
    h = [0 for _ in range(10000001)]
    
    for t in tangerine:
        h[t] += 1
        
    h.sort(reverse=True)
    
    sum = 0
    for i, p in enumerate(h):
        sum += p
        if sum >= k:
            answer = i + 1
            break
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        
        for (int t : tangerine) {
            hmap.put(t, hmap.getOrDefault(t, 0) + 1);
        }
        
        
        Collection<Integer> map = hmap.values();
        List<Integer> arr = new ArrayList<>(map);
        Collections.sort(arr,Collections.reverseOrder());
        
        for (int i : arr) {
            if(k <= 0) {
                break;
            }
            k -= i;
            answer++;
        }
        
        return answer;
    }
}